code/modules/mob/living/carbon/xenomorph/castes/warrior/abilities_warrior.dm

```diff
@@ -150,29 +163,40 @@
 		return FALSE
 	if(!A)
 		return FALSE
+
 	if(!owner.Adjacent(A))
 		return FALSE
-	if(!ishuman(A))
-		return FALSE
-	var/mob/living/carbon/human/H = A
-	if(H.stat == DEAD)
+
+	if(!isliving(A))
 		return FALSE
 
 /datum/action/xeno_action/activable/fling/use_ability(atom/A)
 	var/mob/living/carbon/xenomorph/X = owner
-	var/mob/living/carbon/human/H = A
+	var/mob/living/victim = A
+	var/facing = get_dir(X, victim)
+	var/fling_distance = 4
+	var/stun_duration = 0.5 SECONDS
+
+	X.face_atom(victim) //Face towards the victim
+
 	GLOB.round_statistics.warrior_flings++
 	SSblackbox.record_feedback("tally", "round_statistics", 1, "warrior_flings")
 
-	X.visible_message("<span class='xenowarning'>\The [X] effortlessly flings [H] to the side!</span>", \
-	"<span class='xenowarning'>We effortlessly fling [H] to the side!</span>")
-	playsound(H,'sound/weapons/alien_claw_block.ogg', 75, 1)
-	succeed_activate()
-	H.apply_effects(1,1) 	// Stun
-	shake_camera(H, 2, 1)
+	X.visible_message("<span class='xenowarning'>\The [X] effortlessly flings [victim] away!</span>", \
+	"<span class='xenowarning'>We effortlessly fling [victim] away!</span>")
+	playsound(victim,'sound/weapons/alien_claw_block.ogg', 75, 1)
+
+	if(victim.mob_size >= MOB_SIZE_BIG) //Penalize fling distance for big creatures
+		fling_distance = FLOOR(fling_distance * 0.5, 1)
+
+	if(isxeno(victim))
+		var/mob/living/carbon/xenomorph/x_victim
+		if(X.issamexenohive(x_victim)) //We don't fuck up friendlies
+			stun_duration = 0
+
+	victim.ParalyzeNoChain(stun_duration)
```

Comment:
Just have it early return or stun no Xenos instead of having the stun duration var